tensorbay/opendataset/BDD100K/loader.py

```diff
@@ -300,47 +312,120 @@ def _BDD100K_MOTS2020(path: str) -> Dataset:
     dataset.notes.is_continuous = True
-    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog_mots.json"))
-    _load_mots_segment(dataset, root_path)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), f"catalog_{tracking_type}.json"))
+    images_directory = os.path.join(root_path, "images", _TRACKING_DATASET_INFO[tracking_type][1])
+    labels_directory = os.path.join(root_path, "labels", _TRACKING_DATASET_INFO[tracking_type][2])
+    _load_tracking_segment(dataset, images_directory, labels_directory, load_label)
 
     return dataset
 
 
-def _load_mots_segment(dataset: Dataset, root_path: str) -> None:
-    images_directory = os.path.join(root_path, "images", "seg_track_20")
-    labels_directory = os.path.join(root_path, "labels", "seg_track_20", "polygons")
+def _load_tracking_segment(
+    dataset: Dataset,
+    images_directory: str,
+    labels_directory: str,
+    load_label: Callable[[Dict[str, Any], Data], None],
+) -> None:
+
     for segment_prefix in _SEGMENT_NAMES:
         image_directory = _utility.glob(os.path.join(images_directory, segment_prefix, "*"))
-
+        labels_directory_segment = os.path.join(labels_directory, segment_prefix)
         if segment_prefix == "test":
-            for image_subdir in image_directory:
-                segment = dataset.create_segment(
-                    f"{segment_prefix}_{os.path.basename(image_subdir)}"
-                )
-                for image_path in _utility.glob(os.path.join(image_subdir, "*.jpg")):
-                    segment.append(Data(image_path))
+            load_segment: Callable[
+                [Any, str, str, Callable[[Dict[str, Any], Any], None]], None
+            ] = _load_test_segment
         else:
-            labels_directory_segment = os.path.join(labels_directory, segment_prefix)
-            for image_subdir in image_directory:
-                segment = dataset.create_segment(
-                    f"{segment_prefix}_{os.path.basename(image_subdir)}"
-                )
-                label_filename = f"{os.path.basename(image_subdir)}.json"
-                _load_mots_data(
-                    image_subdir,
-                    os.path.join(labels_directory_segment, label_filename),
-                    segment,
-                )
-
-
-def _load_mots_data(image_subdir: str, label_path: str, target_segment: Segment) -> None:
+            load_segment = _load_other_segment
+        for image_subdir in image_directory:
+            segment = dataset.create_segment(f"{segment_prefix}_{os.path.basename(image_subdir)}")
+            load_segment(segment, image_subdir, labels_directory_segment, load_label)
+
+
+def _load_test_segment(
+    segment: Segment, image_subdir: str, _: str, __: Callable[[Dict[str, Any], Data], None]
+) -> None:
+    for image_path in _utility.glob(os.path.join(image_subdir, "*.jpg")):
+        segment.append(Data(image_path))
+
+
+def _load_other_segment(
```

Comment:
since we does not need unlabeled segment any more, better to yield Data from the function.